src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -712,20 +723,32 @@ def _visit_indirect_addressing(self, node: itir.FunCall) -> IteratorExpr:
             args = [
                 ValueExpr(conn, table.table.dtype),
                 ValueExpr(iterator.indices[shifted_dim], dace.int64),
+                ValueExpr(element_node, dace.int64),
             ]
 
             internals = [f"{arg.value.data}_v" for arg in args]
-            expr = f"{internals[0]}[{internals[1]}, {element}]"
-        else:
+            expr = f"{internals[0]}[{internals[1]}, {internals[2]}]"
+        elif isinstance(self.offset_provider[offset], StridedNeighborOffsetProvider):
             offset_provider = self.offset_provider[offset]
-            assert isinstance(offset_provider, StridedNeighborOffsetProvider)
 
             shifted_dim = offset_provider.origin_axis.value
             target_dim = offset_provider.neighbor_axis.value
             offset_value = iterator.indices[shifted_dim]
-            args = [ValueExpr(offset_value, dace.int64)]
-            internals = [f"{offset_value.data}_v"]
-            expr = f"{internals[0]} * {offset_provider.max_neighbors} + {element}"
+            args = [
+                ValueExpr(offset_value, dace.int64),
+                ValueExpr(element_node, dace.int64),
+            ]
+            internals = [f"{arg.value.data}_v" for arg in args]
+            expr = f"{internals[0]} * {offset_provider.max_neighbors} + {internals[1]}"
+        else:
+            assert isinstance(self.offset_provider[offset], Dimension)
+            shifted_dim = target_dim = self.offset_provider[offset].value
+            args = [
+                ValueExpr(iterator.indices[shifted_dim], dace.int64),
+                element_node,
+            ]
+            internals = [f"{arg.value.data}_v" for arg in args]
+            expr = f"{internals[0]} + {internals[1]}"
```

Comment:
It seems to me that this part of the code duplicates the functionality of `_visit_direct_addressing`. The dispatch between the two is made line 488. Due to the dispatch, this code will never actually be called.

You can leave all the different offset providers in one function, like this, delete the `_visit_direct_addressing`, and remove the dispatch, or you can add a `_visit_strided_addressing` function and extend the dispatch. Both would solve the issue.